pkg/network/ebpf/c/co-re/tracer-fentry.c

```diff
@@ -279,8 +279,40 @@ int BPF_PROG(udpv6_recvmsg_exit, struct sock *sk, struct msghdr *msg, size_t len
 
 SEC("fentry/tcp_retransmit_skb")
 int BPF_PROG(tcp_retransmit_skb, struct sock *sk, struct sk_buff *skb, int segs, int err) {
-    log_debug("fexit/tcp_retransmit: segs: %d\n", segs);
-    return handle_retransmit(sk, segs);
+    u64 tid = bpf_get_current_pid_tgid();
+    tcp_retransmit_skb_args_t args = {};
+    args.sk = sk;
+    args.segs = 0;
+    args.retrans_out_pre = 0;
+    if (bpf_probe_read_kernel_with_telemetry(&(args.retrans_out_pre), sizeof(args.retrans_out_pre), &(tcp_sk(sk)->retrans_out)) < 0) {
```

Comment:
Use `BPF_CORE_READ` or `BPF_CORE_READ_INTO` here.

Reply:
Lots of details here: https://nakryiko.com/posts/bpf-core-reference-guide/#reading-kernel-data

tl;dr the macros make using `bpf_core_read` easier to use. `bpf_core_read` should be used here instead of `bpf_probe_read` since we want to record information about the field offsets so that they can be adjusted at runtime.